platform/netbinox/src/org/netbeans/modules/netbinox/NetbinoxFactory.java

```diff
@@ -74,6 +77,45 @@ public Framework newFramework(Map map) {
             configMap.put("osgi.locking", "none"); // NOI18N
         }
 
+        // Ensure that the org.osgi.framework.executionenvironment holds all
+        // JavaSE entries that match till the current JDK. The dynamic approach
+        // will work also for newly released JDKs.
+        String[] javaSpecificationVersion = System.getProperty("java.specification.version").split("\\.");
+        Integer javaSpecificationMajorVersion = null;
+        try {
+            javaSpecificationMajorVersion = Integer.valueOf(javaSpecificationVersion[0]);
+        } catch (NumberFormatException ex) {
+            LOG.log(
+                    Level.WARNING,
+                    "Failed to parse java.specification.version: {0}",
+                    System.getProperty("java.specification.version")
+            );
+        }
```

Comment:
since this needs to work on 9 and later (the other versions are hardcoded), we could use the API here to be future proof:

```java
       try {
            // java >9
            Object runtimeVersion = Runtime.class.getMethod("version").invoke(null);
            javaSpecificationMajorVersion = (int) runtimeVersion.getClass().getMethod("major").invoke(runtimeVersion);
       } catch (ReflectiveOperationException ignore) {
            // java <=9
       }
```
using `major` instead of `feature` is deprecated but not for removal and it is returning the same value.
https://docs.oracle.com/javase/9/docs/api/java/lang/Runtime.Version.html
https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Runtime.Version.html#major()

Reply:
Thank you for the idea. I updated the PR in place.

Reply:
@matthiasblaesing I think you have to set javaSpecificationMajorVersion to 8 in the catch, otherwise the if will not run if its run on JDK 8 since it has a >9 check.

Reply:
On JDKs < 9 the default behavior of Equinox takes over, which today always active.

Reply:
ok, understood